### DISCUSSION TITLE
Auto scale keyframe placement when changing an animation's fps

### DISCUSSION THREAD

**Discussion by Videonoize** [godotengine/godot-proposals#11205] [Editor]
Currently when you change the fps of an animation, the keyframes do not adjust with the new frame rate markers. If I change an animation from 14 to 12 fps, I have to manually drag each individual keyframe, of each element of the animation, onto the new frame markers. This makes adjusting animation frame rates a very tedious process.

**Comment by Calinou** [comment]
This should be a manual operation, not something that happens automatically every time you change the snapping distance (which is either configured in time or FPS, but FPS is just another way to express `1.0 / time`).

**Reply by Videonoize** [reply]
Sorry, I don't understand what you mean by this should be a manual process, why would that be the case? To me it seems like a very unnecessarily tedious process to make changes without this very simple fix.

**Reply by pixel-boy** [reply]
The idea of ​​being able to adapt the size of an animation in an AnimationPlayer would be practical, it should not necessarily be the default option, but allowing to do it in another way would clearly be an improvement for the workflow.
